cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,88 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SettingsPageObject from '../support/pages/settings.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import userPageObject from '../support/pages/user.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject';
+
+const settingsPage = new SettingsPageObject();
+const homePage = new HomePageObject();
+const userPage = new userPageObject();
+const signInPage = new SignInPageObject();
+
+const testData = {
+  newUsername: 'edited_username',
+  newBio:'edited_bio',
+  newEmail: 'edited_email@qa.team',
+  newPassword: 'Edited_password1!'
+}
+
 describe('Settings page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
+    cy.task('db:clear');
+    cy.wait(500);
 
+    cy.login(user.email, user.username, user.password);
+    settingsPage.visit();
   });
 
   it('should provide an ability to update username', () => {
-
+    settingsPage.editUsernameInput.clear().type(testData.newUsername);
+    settingsPage.updateBtn.click();
+    settingsPage.assertEditing('Update successful!');
+    homePage.usernameLink.should('contain', testData.newUsername);
   });
 
   it('should provide an ability to update bio', () => {
+    settingsPage.editBioInput.clear().type(testData.newBio);
+    settingsPage.updateBtn.click();
 
+    settingsPage.assertEditing('Update successful!');
+    userPage.visitUserPage(user.username);
+    userPage.profileBio.should('contain', testData.newBio);
   });
 
-  it('should provide an ability to update an email', () => {
+  it.skip('should provide an ability to update an email', () => {
```

Comment:
This test has been skipped. If this was intentional, please ignore this. Otherwise, you should remove '.skip' to allow the test to run.